libraries/drivers_appserver_puma.rb

```diff
@@ -33,6 +34,30 @@ def appserver_config
         'puma.rb'
       end
 
+      # Invoke the monit start command for the appserver. This may only be
+      # needed during the initial setup of the instance. After that the
+      # 'restart' command is sufficient. If puma is already running this
+      # resource will not do anything.
+      def start_monit
+        pidfile = "/var/run/lock/#{app['shortname']}/puma.pid"
+        context.execute "monit start #{adapter}_#{app['shortname']}" do
+          retries 3
+          creates pidfile
+        end
+      end
+
+      # Immediately attempts to restart the appserver using monit. Do not
+      # attempt a restart if the pid file doesn't exist
+      def restart_monit
+        return if ENV['TEST_KITCHEN'] # Don't like it, but we can't run multiple processes in Docker on travis
+
+        pidfile = "/var/run/lock/#{app['shortname']}/puma.pid"
+        context.execute "monit restart #{adapter}_#{app['shortname']}" do
+          retries 3
+          only_if { ::File.exist?(pidfile) }
```

Comment:
Only runs if there is a pidfile, which generally means that puma is already running.